### PR TITLE
[Key Vault] removing the keyvault storage commands from the 2016-10-01 profile

### PR DESCRIPTION
**Pull request by schaabs**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by bganapa** [comment]
Ack for removing, azurestack  does not support this  and Per scott, this was not supposed to exposed thru 2016-10-01 api-version

**Comment by troydai** [comment]
@tjprescott please merge if it looks good to you.

File: src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/commands.py [hunk 1]
```diff
@@ -134,14 +134,14 @@ def load_command_table(self, _):
...
-    with self.command_group('keyvault storage', kv_data_sdk) as g:
-        g.keyvault_command('add', 'set_storage_account')
-        g.keyvault_command('list', 'get_storage_accounts')
-        g.keyvault_command('show', 'get_storage_account')
-        g.keyvault_command('update', 'update_storage_account')
-        g.keyvault_command('remove', 'delete_storage_account')
-        g.keyvault_command('regenerate-key', 'regenerate_storage_account_key')
-        if data_api_version != '2016_10_01':
+    if data_api_version != '2016_10_01':
+        with self.command_group('keyvault storage', kv_data_sdk) as g:
```
Comment (tjprescott) [inline src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/commands.py]: This should use that standard syntax for specifying that commands only apply to a certain API version.

https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#keyword-argument-reference

File: src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/commands.py [hunk 2]
```diff
@@ -153,17 +153,17 @@ def load_command_table(self, _):
...
-                           doc_string_source=data_doc_string.format('set_sas_definition'))
-        g.keyvault_command('list', 'get_sas_definitions')
-        g.keyvault_command('show', 'get_sas_definition')
-        g.keyvault_command('update',
-                           'update_sas_definition',
-                           doc_string_source=data_doc_string.format('update_sas_definition'))
-        g.keyvault_command('delete', 'delete_sas_definition')
-        if data_api_version != '2016_10_01':
+    if data_api_version != '2016_10_01':
+        with self.command_group('keyvault storage sas-definition', kv_data_sdk) as g:
```
Comment (tjprescott) [inline src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/commands.py]: Same here. The command_group call should include the min_api kwarg to effect this change.

**Comment by troydai** [comment]
Ping @schaabs

**Comment by troydai** [comment]
@schaabs

File: src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/commands.py (hunk 1 above)
Comment (schaabs) [reply src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/commands.py]: @tjprescott I tested setting min_api='7.0' for the command groups, however this results in the commands being filtered even when running on the latest profile.  Does the min_api parameter support filtering for our data-plane version? or only the management plane api version?

**Comment by schaabs** [comment]
@troydai @tjprescott Please see my response to the comment above.

File: src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/commands.py (hunk 1 above)
Comment (bganapa) [reply src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/commands.py]: min_api implementation expects the api version to be in date time format.. implementation here https://github.com/Azure/azure-cli/blob/ce4fac89bad98266ad6701e52c76771735775946/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L255

I think we should not have allowed 7.0 to be a version in the SDK level itself and standardize with date time format everywhere.

Comment (tjprescott) [reply src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/commands.py]: @lmazuel implemented the change in CLI to support version comparisons between dates and arbitrary numbers. Laurent, can you take a look?

Comment (tjprescott) [reply src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/commands.py]: If something needs to be fixed, it's the version comparison logic. We shouldn't allow this nonstandard way of filtering the commands.

Comment (lmazuel) [reply src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/commands.py]: So, it goes deeper than I thought. I recommend to put max_api='42.0' for now (aka a value so high that's it's not a problem for the next century), and to discuss more deeply some design concern I have with Profile implemetation. Especially, the fact that profile name carries a semantic or not (actually they don't, but CLI expects they do and parse the name)

**Comment by tjprescott** [comment]
This may be a brittle solution but if we need to revisit the concept of profiles it is as good as it can be right now.

**Comment by troydai** [comment]
Resolve the conflict manually. Merge once CI passed.

**Comment by troydai** [comment]
I will merge and fix the CI as the setup file needs to be updated.

**Pull request merged**